crates/sui-indexer/src/models_v2/events.rs

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::str::FromStr;
+
+use diesel::prelude::*;
+use move_bytecode_utils::module_cache::GetModule;
+use move_core_types::identifier::Identifier;
+use move_core_types::value::MoveStruct;
+
+use sui_json_rpc_types::{SuiEvent, SuiMoveStruct};
+use sui_types::base_types::{ObjectID, SuiAddress};
+use sui_types::event::EventID;
+use sui_types::object::{MoveObject, ObjectFormatOptions};
+use sui_types::parse_sui_struct_tag;
+
+use crate::errors::IndexerError;
+use crate::schema_v2::events;
+use crate::types_v2::IndexedEvent;
+
+#[derive(Queryable, Insertable, Debug, Clone)]
+#[diesel(table_name = events)]
+pub struct StoredEvent {
+    pub tx_sequence_number: i64,
+    pub event_sequence_number: i64,
+    pub transaction_digest: Vec<u8>,
+    pub checkpoint_sequence_number: i64,
+    pub senders: Vec<Vec<u8>>,
+    pub package: Vec<u8>,
+    pub module: String,
+    pub event_type: String,
```

Comment:
The asymmetry is interesting here -- why do we store the calling package and module separately, but the event type (which is a triple of package, module, type name) as one column?

Reply:
(Answering my own question, maybe it's because the type can have type parameters).